pkg/berglas/deleter.go

```diff
@@ -58,6 +62,33 @@ func (c *Client) Delete(ctx context.Context, i *DeleteRequest) error {
 		return errors.New("missing object name")
 	}
 
+	if i.Permanently {
+		it := c.storageClient.
+			Bucket(bucket).
+			Objects(ctx, &storage.Query{
+				Prefix:   object,
+				Versions: true,
+			})
+
+		for {
```

Comment:
Having [been down this road before](https://github.com/terraform-providers/terraform-provider-google/pull/2944), you're going to want to parallelize deletion. github.com/gammazero/workerpool is usually my go-to.

Reply:
Btw, since the bucket policy restricts up to 10 versions only - do you think this is still required? Related question: should berglas support buckets not created by the tool? Should bucket configuration be overridden for buckets not created by berglas but are used by it? (there could be warning etc..)

Reply:
I think so:

1. There are existing berglas installs out there without versioning policy
1. Users can change the policy